modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvPartitionStorageGcTest.java

```diff
@@ -129,4 +129,17 @@ void testManyOldVersions() {
         // Nothing else to poll.
         assertNull(pollForVacuum(lowWatermark));
     }
+
+    @Test
+    void testTombstoneFirst() {
+        addAndCommit(null);
+
+        addAndCommit(TABLE_ROW);
+
+        addAndCommit(TABLE_ROW2);
+
+        BinaryRowAndRowId row = pollForVacuum(HybridTimestamp.MAX_VALUE);
+
+        assertNull(row.binaryRow());
```

Comment:
Wow, did not expect that! I thought it should be TABLE_ROW.
Why did you choose such weird behavior? Very non-intuitive